src/vscode-bicep/src/test/e2e/hover.test.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as vscode from "vscode";
+
+import {
+  expectDefined,
+  expectRange,
+  marked,
+  normalizeMarkedString,
+  readExampleFile,
+} from "../utils";
+import { executeHoverProviderCommand } from "./commands";
+
+describe("hover", (): void => {
+  let document: vscode.TextDocument;
+
+  beforeAll(async () => {
+    const content = readExampleFile("101", "vm-simple-linux");
+    document = await vscode.workspace.openTextDocument({
+      language: "bicep",
+      content,
+    });
+
+    await vscode.window.showTextDocument(document);
+  });
+
+  afterAll(async () => {
+    await vscode.commands.executeCommand("workbench.action.closeActiveEditor");
+  });
+
+  it("should reveal type signature when hovering over a parameter name", async () => {
+    const hovers = await executeHoverProviderCommand(
+      document.uri,
+      new vscode.Position(1, 7)
+    );
+
+    expectHovers(hovers, {
+      startLine: 1,
+      startCharacter: 6,
+      endLine: 1,
+      endCharacter: 12,
+      contents: ["param vmName: string"],
+    });
```

Comment:
Note these examples are liable to change; tightly coupling the tests to them may make changing them difficult (especially as example contributors aren't necessarily familiar with this codebase). Is it worth adding sample `.bicep` files directly to the `src/vscode-bicep/src/test/e2e` folder to avoid this?

Reply:
> especially as example contributors aren't necessarily familiar with this codebase

Yeah this makes sense. I'll add samples to the e2e folder.

Reply:
Done.